synapse/http/server.py

```diff
@@ -254,7 +209,101 @@ def register_paths(self, method, path_patterns, callback):
         pass
 
 
-class JsonResource(HttpServer, resource.Resource):
+class _AsyncResource(resource.Resource, metaclass=abc.ABCMeta):
+    """Base class for resources that have async handlers.
+
```

Comment:
a word or two about how it works and what the subclasses are meant to override would be helpful